mycroft/util/memoize.py

```diff
@@ -0,0 +1,40 @@
+import collections
+import functools
+
+from .log import LOG
+
+
+# The following memoized class is copied from the PythonDecoratorLibrary at:
```

Comment:
Difference between built in lru_cache built in?

https://docs.python.org/dev/library/functools.html#functools.lru_cache

We use it in padatious_service